public/app/plugins/datasource/tempo/NativeSearch/NativeSearch.tsx

```diff
@@ -52,7 +51,7 @@ const NativeSearch = ({ datasource, query, onChange, onBlur, onRunQuery }: Props
         if (isFetchError(error) && error?.status === 404) {
           setError(error);
         } else if (error instanceof Error) {
-          dispatch(notifyApp(createErrorNotification('Error', error)));
+          setAlertText(`Error: ${error.message}`);
```

Comment:
Seems a bit weird to me that here and also in other places we have `setError` and `setAlertText` at the same time. Should that be a single thing?

Reply:
Also I think we need to be reseting the error state at some point. Based on how the TemporaryAlert works it seems like it becomes visible again only if severity or text changes. Imagine a situation you do a query here, you get some error, then it works for a while but the you get the same error again. As we didn't reset the state in between we won't show the later error because the test of the error didn't change.

Reply:
> `setError` and `setAlertText`

I was also thinking to normalize the errors. It's not clear to me why it was like that even before—these changes are just replacing the `dispatch` with the `setAlertText`, but we had that problem even before. The main difference is that `setError` is used to show an error inside the trace tables, while `dispatch` (now `setAlertText`) to show a floating alert.